### DISCUSSION TITLE
how to understand ActionModelImpulseFwdDynamics?

### DISCUSSION THREAD

**Discussion by liuyidesktop** [loco-3d/crocoddyl#1270] [General]
1.I noticed that in the example of the walking program, ActionModelImpulseFwdDynamics is used to describe impulse dynamics. What is the purpose of this part?
2.I found that the input at the ActionModelImpulseFwdDynamics stage does not exist in the obtained solution, and the state only involves changes in velocity. So, is this an instantaneous process without any integration step?
3.How should this point be handled in MPC control? I noticed that removing this step significantly increases the number of iterations.

**Comment by cmastalli** [comment]
Hi @liuxiaoyiyixiaoliu,

The answers to your questions are:
  1. Impulse dynamics act as reset maps in hybrid systems. I suggest learning more about the formalism of hybrid systems.
  2. Yes, it can be seen as an instantaneous process or jump in the state of the system.
  3. As we work with hybrid systems, we need to formulate a hybrid optimal control problem (or MPC). Removing impulse dynamics increases the number of iterations, especially in jumps. Have you checked everything carefully?

**Reply by liuyidesktop** [reply]
OK, thanks for your answer
